### ISSUE TITLE
debian wheezy mips builds please

### ISSUE THREAD

**Issue by syadnom** [zerotier/ZeroTierOne#702]
Target is the edgerouter platform from ubiquiti.  These run a MIPS 1004Kc chip with isa mips32r1/r2.  

It would be great if the debian zerotier install sh were able to install on these.

**Comment by adamierymenko** [comment]
Wheezy is old... really old! There are no C++11 capable compiler packages for it. Right now to support Wheezy we build static binaries in Alpine since musl-libc supports static builds. These are used for non-standard and old Linux distributions. Unfortunately there is no mips architecture for Alpine.

Is it mips or mipsel (little-endian)?

We could investigate but I can't make any promises. If it's too much work we'll have to punt on this one as both wheezy and 32-bit MIPS are quite old and don't have substantial user bases.

**Comment by adamierymenko** [comment]
Going to see if jessie, stretch, or buster can build for this, otherwise wontfix.

**Comment by adamierymenko** [comment]
We now have mips for several Debian variants in our build farm, which will be used for 1.2.6 and later.

**Comment by lunarthegrey** [comment]
Is it just me or do I not see any mips .debs in the repo?

**Comment by lunarthegrey** [comment]
Just tried it out on a mips device.

```
ubnt@erx-home-01:~$ curl -s https://install.zerotier.com/ | sudo bash

*** ZeroTier One Quick Install for Unix-like Systems

*** Tested distributions and architectures:
***   MacOS (10.7+) on x86_64 (just installs ZeroTier One.pkg)
***   Debian (7+) on x86_64, x86, arm, and arm64
***   RedHat/CentOS (6+) on x86_64 and x86
***   Fedora (16+) on x86_64 and x86
***   SuSE (12+) on x86_64 and x86
***   Mint (18+) on x86_64, x86, arm, and arm64

*** Please report problems to contact@zerotier.com and we will try to fix.

*** Detecting Linux Distribution

*** Found Debian "wheezy" (or similar), creating /etc/apt/sources.list.d/zerotier.list
OK

*** Installing zerotier-one package...
Get:1 http://download.zerotier.com wheezy Release.gpg [800 B]
Get:2 http://download.zerotier.com wheezy Release [6465 B]
Fetched 7265 B in 0s (9945 B/s)    
W: Failed to fetch http://download.zerotier.com/debian/wheezy/dists/wheezy/Release  Unable to find expected entry 'main/binary-mipsel/Packages' in Release file (Wrong sources.list entry or malformed file)

E: Some index files failed to download. They have been ignored, or old ones used instead.
Reading package lists... Done
Building dependency tree... Done
E: Unable to locate package zerotier-one

*** Package installation failed! Unfortunately there may not be a package
*** for your architecture or distribution. For the source go to:
*** https://github.com/zerotier/ZeroTierOne
```

**Comment by glimberg** [comment]
@lunarthegrey To quote from above:

> Wheezy is old... really old! There are no C++11 capable compiler packages for it.

ZeroTier requires C++11.  No C++11, no ZeroTier One

**Comment by glimberg** [comment]
Also, on *some* older platforms, we can statically link in the libraries we need with a QEMU chroot system we have based on Alpine Linux, but Alpine doesn't support MIPS so we can't build it on there with C++11 either.

**Comment by jfdhuiz** [comment]
Seems zt1 1.2.12 is available for Wheezy, but only for amd64, i386, and armhf. Which are the most popular platforms, but means ER-X (mips32el) and ER-L (mips64) are missing out.

https://download.zerotier.com/debian/wheezy/pool/main/z/zerotier-one/

On my Synology machine, I got zt1 working but not wg.

On my ER-L, I got wg working but not zt1.

The last posts in this thread are by the person who does the wg edgeos ports

https://bugs.alpinelinux.org/issues/5643

He did statically link the musl library, but I'm not sure he used Alpine.
